keras/backend/tensorflow_backend.py

```diff
@@ -3638,3 +3638,46 @@ def foldr(fn, elems, initializer=None, name=None):
         Same type and shape as initializer
     """
     return tf.foldr(fn, elems, initializer=initializer, name=name)
+
+
+def dropout_on_constant_mask(inputs, dim, dropout_value, training):
+    """Generate a drop out mask as the constant during recurrent.
```

Comment:
These two new methods introduced is because cntk's way to handle the drop out mask is a little different as the existing keras code, so we did a small refactoring to avoid add "if K.backend() == cntk" statement in keras recurrent layer. There is no new feature in tensorflow_backend and theano_backend.  Do we prefer to have the refactoring of recurrent layer with the two new method in a separate PR?